geopandas/geodataframe.py

```diff
@@ -19,6 +20,11 @@
 from . import _compat as compat
 from ._decorator import doc
 
+if compat.SHAPELY_GE_18:
+    geometry_type_error = shapely.errors.GeometryTypeError
+else:
+    geometry_type_error = ValueError
```

Comment:
Is older shapely returning a ValueError or a TypeError? I don't have any env with shapely 1.7 to test but this feels strange (though we may not care much about 1.7 now).

Reply:
Yeah, I would have expected a typeerror too if anything, but it's not.
 https://github.com/shapely/shapely/blob/28b748f35ec0483da7bd8fc64b65ee5eee0aff1d/shapely/geometry/geo.py#L115-L116
(it also seems a bit weird to conditionally define the extra exception, rather than inline something with `compat.SHAPELY_GE_18`, but I guess it's fine)